### PR TITLE
[3006.x] Fix install of logrotate config for debian pkg

### PR DESCRIPTION
**Pull request by barneysowood**
### What does this PR do?

Fixes the install of logrotate config for the debian pkg by moving file to pkg/common/logrotate/salt-common. File is installed via salt-common.install file, which can't rename files, only copy them to a directory, so we need to rename the file and put it in a subdir of pkg/common.

If the incorrect /etc/logrotate.d/salt directory created by original 3006.3 packages is present, it is removed as part of preinst.

Adds /etc/logrotate.d/salt-common to salt.common.conffiles to ensure that dpkg will not overwrite configs modified by users.

Also updates RPM spec file for new location of logrotate config.

Config needs to be /etc/logrotate.d/salt-common as that is what is used by 3005.x packages - if we don't use the same name, we'll end up with duplicate/conflicting configs.

### What issues does this PR fix or reference?
Fixes: #65231

### Previous Behavior
Logrotate config was incorrectly copied to /etc/logrotated.d/salt/salt-common.logrotate

### New Behavior
Config is copied to /etc/logrotate.d/salt-common and marked as a config file so dpkg will not overwrite modified configs.

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [x] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [x] Tests written/updated

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Comment by barneysowood** [comment]
@s0undt3ch and @dmurphy18 - could I get a review of this? Thanks!

**Review by dmurphy18 — DISMISSED** [review]
So long as you fix the issue @terminalmage raised about salt-common

**Comment by terminalmage** [comment]
> So long as you fix the issue @terminalmage raised about salt-common

See here: https://github.com/saltstack/salt/issues/65231#issuecomment-1785708839

File: pkg/tests/integration/test_logrotate_config.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
+"""
+Tests for logrotate config
+"""
+
+from pathlib import Path
+
+import packaging.version
+import pytest
+
+pytestmark = [pytest.mark.skip_on_windows, pytest.mark.skip_on_darwin]
```
Comment (s0undt3ch) [inline pkg/tests/integration/test_logrotate_config.py:10]:
```suggestion
pytestmark = [
    pytest.mark.skip_unless_on_linux,
]
```

File: pkg/tests/integration/test_logrotate_config.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    """
+    logrotate_dir = Path("/etc/logrotate.d")
+
+    os_family = salt_call_cli.run("grains.get", "os_family")
+    assert os_family.returncode == 0
+
+    if os_family.data == "RedHat":
+        return logrotate_dir / "salt"
+    elif os_family.data == "Debian":
+        return logrotate_dir / "salt-common"
```
Comment (s0undt3ch) [inline pkg/tests/integration/test_logrotate_config.py:26]:
```suggestion
@pytest.fixture
def logrotate_config_file(grains):
    """
    Fixture for logrotate config file path
    """
    logrotate_dir = pathlib.Path("/etc/logrotate.d")

    if grains["os_family"] == "RedHat":
        return logrotate_dir / "salt"
    elif grains["os_family"] == "Debian":
        return logrotate_dir / "salt-common"
```

File: pkg/tests/integration/test_logrotate_config.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
+"""
+Tests for logrotate config
+"""
+
+from pathlib import Path
```
Comment (s0undt3ch) [inline pkg/tests/integration/test_logrotate_config.py:5]:
```suggestion
import pathlib
```

File: pkg/tests/integration/test_logrotate_config.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+def test_issue_65231_etc_logrotate_salt_dir_removed(install_salt):
+    """
+    Test that /etc/logrotate.d/salt is not a directory
+    """
+    if install_salt.prev_version and packaging.version.parse(
+        install_salt.prev_version
+    ) == packaging.version.parse("3006.3"):
+        pytest.skip("Testing a downgrade to 3006.3, do not run")
+
+    path = Path("/etc/logrotate.d/salt")
```
Comment (s0undt3ch) [inline pkg/tests/integration/test_logrotate_config.py:47]:
```suggestion
    path = pathlib.Path("/etc/logrotate.d/salt")
```

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

**Review by s0undt3ch — DISMISSED** [review]

**Review by s0undt3ch — APPROVED** [review]

**Pull request merged**